### DISCUSSION TITLE
Deprecate Newtonsoft?

### DISCUSSION THREAD

**Discussion by TonyValenti** [microsoft/vs-streamjsonrpc#916] [General]
Is it possible to use this library without newtonsoft?  Id like to stick to STJ.

**Comment by AArnott** [comment]
You can use the `SystemTextJsonFormatter` instead of the `JsonMessageFormatter` to avoid Newtonsoft, yes. 
I think you still need to ship Newtonsoft.Json.dll with your application, as the CLR will try to load it when doing reflection on attributes that are in this library, but it wouldn't use it for serialization.

Note that System.Text.Json support is very new (#305) so it will first release in v2.16 of this library.
